### ISSUE TITLE
Add Qistkit Nature demonstrations for physics problems

### ISSUE THREAD

**Issue by e-eight** [qiskit-community/mentorship-program-fall-21#27]
# Description
Currently most of the tutorials in Qiskit Nature deal with solving problems in quantum chemistry, and they are very much tied to the use of molecular Hamiltonians, and quantum chemistry frameworks like PySCF. However there are problems in quantum physics which can be solved using the same quantum computing principles. Some examples can be found at https://arxiv.org/abs/1801.03897, https://arxiv.org/abs/2105.10941 and https://arxiv.org/abs/2011.13443. These do not use any molecular Hamiltonians but demonstrate the use of quantum algorithms for solving quantum physics problems. It will be nice to develop tutorials or demonstrations showing how to efficiently use Qiskit Nature to solve such problems. This will help to expand the use cases of Qiskit Nature and might also lead to adding some new features to Qiskit Nature or streamlining the existing ones.

# Mentor/s
No.

# Type of participant
- Experience with Qiskit and Python.
- Prior knowledge of quantum physics and Qiskit Nature will be helpful.

# Number of participants
1 - 2

# Deliverable
Jupyter notebooks that demonstrate the problem and the quantum solution(s).

**Comment by HuangJunye** [comment]
@e-eight Do you want to co-mentor this project or are you looking for a mentor? In either case, @woodsp-ibm may be interested. Can you two discuss over here or on Slack about this project?

**Comment by e-eight** [comment]
Yes, I would like to co-mentor this project. I will discuss with @woodsp-ibm.

**Comment by SiddharthaMorales** [comment]
I have two masters' in theoretical physics. I'd like to contribute in this project

**Comment by paniash** [comment]
Hi! I'm interested in contributing to this project!

**Comment by shil-m** [comment]
I am a physics major; this project will help me expand my knowledge in Qiskit Nature.

**Comment by HuangJunye** [comment]
@woodsp-ibm @SiddharthaMorales @jvscursulim @paniash Can you please comment in the issue so that I can assign you?

**Comment by woodsp-ibm** [comment]
> @woodsp-ibm @SiddharthaMorales @jvscursulim @paniash Can you please comment in the issue so that I can assign you?

👍

**Comment by SiddharthaMorales** [comment]
Hello :)

**Comment by jvscursulim** [comment]
Hello Quantum World :)

**Comment by HuangJunye** [comment]
@SiddharthaMorales @jvscursulim Can you please upload your presentation here by the end of today? Thank you!

**Comment by jvscursulim** [comment]
@HuangJunye We are going to upload the presentation soon, @SiddharthaMorales and I are fixing some details

**Comment by SiddharthaMorales** [comment]
[#27 Add Qiskit Nature demonstrations for physics problems2.0.pdf](https://github.com/qiskit-advocate/qamp-fall-21/files/7297118/27.Add.Qiskit.Nature.demonstrations.for.physics.problems2.0.pdf)
[#27 Add Qiskit Nature demonstrations for physics problems2.0.pptx](https://github.com/qiskit-advocate/qamp-fall-21/files/7297119/27.Add.Qiskit.Nature.demonstrations.for.physics.problems2.0.pptx)

**Comment by jvscursulim** [comment]
Since the last checkpoint of Qiskit advocate mentorship program, I’ve been working on two things, the first one is the improvement of results obtained through VQEProgram for the binding energy between proton and neutron in the deuteron. For that I need to transpile the ansatz circuit in order to prevent some SWAP gate being introduced, once each SWAP gate increases the number of CNOT gates in the circuit by 3 and we know that could introduce a relevant amount of errors in the computation process. 
	The second problem is the computation of ground state energy of the Heisenberg model in some different lattices (line, triangle, square, pentagon, hexagon, five-pointed star and kagome), especially in the kagome lattice. In order to achieve this goal, I had to write a function that builds the Hamiltonian given a graph that represents the lattice. With the Hamiltonians in hands, I use VQE with the include_custom set as True (In this mode VQE uses a qasm simulator that behaves like a state vector simulator), to compute the ground state energies. Until now, the results for line, triangle, square, pentagon, hexagon and five-pointed star lattices are in good agreement with the reference values got through NumPy eigensolver, but I’m facing some problems with the results for one version of the kagome lattice, it seems there is a barren plateaus problem and currently I’m trying to solve this problem. 

![results](https://user-images.githubusercontent.com/66651520/141196468-80870db3-4891-41c1-a0a6-b3e423c13b1e.png)

**Comment by HuangJunye** [comment]
@SiddharthaMorales @jvscursulim Can you please upload the final presentation slides and link your pull requests if any here? Thank you!

**Comment by jvscursulim** [comment]
[#27 Add Qiskit Nature demonstrations for physics problems - December.pdf](https://github.com/qiskit-advocate/qamp-fall-21/files/7902005/27.Add.Qiskit.Nature.demonstrations.for.physics.problems.-.December.pdf)
